zipline/pipeline/engine.py

```diff
@@ -242,8 +242,11 @@ def _mask_and_dates_for_term(self, term, workspace, graph, dates):
         Load mask and mask row labels for term.
         """
         mask = term.mask
-        offset = graph.extra_rows[mask] - graph.extra_rows[term]
-        return workspace[mask][offset:], dates[offset:]
+        mask_offset = graph.extra_rows[mask] - graph.extra_rows[term]
+        dates_offset = (
+            graph.extra_rows[self._root_mask_term] - graph.extra_rows[term]
```

Comment:
Might be worth commenting here that we're comparing against `_root_mask_term` because that's what determines the shape of the top-level dates array.